### PR TITLE
[NNVM/TOPI][OP] gather_nd

### PR DESCRIPTION
**Pull request by icemelon9**
Add gather_nd op into NNVM and Topi.

### CODE REVIEW

File: nnvm/src/top/tensor/transform.cc [hunk 1]
```diff
@@ -1353,5 +1353,110 @@ Examples::
...
+}
+
+inline bool GatherNDInferType(const NodeAttrs &attrs,
+                              std::vector<int> *in_attrs,
+                              std::vector<int> *out_attrs) {
+  CHECK_EQ(in_attrs->size(), 2U);
+  CHECK_EQ(out_attrs->size(), 1U);
+  CHECK_EQ((*in_attrs)[1], kInt32);
+  NNVM_ASSIGN_INPUT_TYPE(attrs, *in_attrs, 0, (*in_attrs)[0]);
+  NNVM_ASSIGN_INPUT_TYPE(attrs, *in_attrs, 1, static_cast<int>(kInt32));
```
Comment (zhreshold) [inline nnvm/src/top/tensor/transform.cc]: what happens if the second input is the same type as first, say float32?

Comment (icemelon9) [reply nnvm/src/top/tensor/transform.cc]: I think this shouldn't happen, similar as take op.

Comment (zhreshold) [reply nnvm/src/top/tensor/transform.cc]: suppose index is the output of another op with type float, what is going to happen now?
say ``sym.gather_nd(a, sym.floor(sym.log(b)))`` ?

Comment (icemelon9) [reply nnvm/src/top/tensor/transform.cc]: New commit can support float type for indices tensor.

**Pull request merged**